src/Microsoft.VisualStudio.Threading.Tests/TplExtensionsTests.cs

```diff
@@ -966,5 +991,47 @@ protected override bool TryExecuteTaskInline(Task task, bool taskWasPreviouslyQu
                 return this.TryExecuteTask(task);
             }
         }
+
+        private class MyValueTaskSource : IValueTaskSource
+        {
+            internal AsyncManualResetEvent GetResultCalled = new AsyncManualResetEvent();
+
+            public void GetResult(short token)
+            {
+                this.GetResultCalled.Set();
+            }
+
+            public ValueTaskSourceStatus GetStatus(short token)
+            {
+                return ValueTaskSourceStatus.Pending;
+            }
+
+            public void OnCompleted(Action<object> continuation, object state, short token, ValueTaskSourceOnCompletedFlags flags)
+            {
+                Task.Factory.StartNew(continuation, state, CancellationToken.None, TaskCreationOptions.None, TaskScheduler.Default).Forget();
+            }
+        }
+
+        private class MyValueTaskSource<T> : IValueTaskSource<T>
+            where T : struct
```

Comment:
If you wanted, you could delete the `MyValueTaskSource` and just have `MyValueTaskSource<T>` implement both interfaces: then you'd only need to implement the two `void GetResult` and `T GetResult` methods, rather than implementing the whole interface twice.

Reply:
I considered that, but was undecided. I'll go for it. Thanks.